File: MATLAB/example_usage.m
% example_usage.m
% Example usage of the RSA encryption and decryption in MATLAB

% Prime numbers
p = 61;
q = 53;

% Generate public and private keys
[e, d, n] = rsa_generate_keys(p, q);
fprintf('Public Key (e, n): (%d, %d)\n', e, n);
fprintf('Private Key (d, n): (%d, %d)\n', d, n);

% Message to encrypt
message = 65; % Example message (A)
fprintf('Original message: %d\n', message);

% Encrypt the message
encrypted_message = rsa_encrypt(message, e, n);
fprintf('Encrypted message: %d\n', encrypted_message);

% Decrypt the message
decrypted_message = rsa_decrypt(encrypted_message, d, n);
fprintf('Decrypted message: %d\n', decrypted_message);

File: C/example_usage.c
// example_usage.c
// Example usage of the RSA encryption and decryption in C

#include <stdio.h>
#include "rsa.c"

int main() {
    int p = 61, q = 53;  // Example prime numbers
    int e, d, n;
    
    // Generate keys
    rsa_generate_keys(&e, &d, &n, p, q);

    // Display public and private keys
    printf("Public Key (e, n): %d, %d\n", e, n);
    printf("Private Key (d, n): %d, %d\n", d, n);

    // Encrypt and decrypt message
    int message = 65; // Example message (A)
    printf("Original message: %d\n", message);

    int encrypted = rsa_encrypt(message, e, n);
    printf("Encrypted message: %d\n", encrypted);

    int decrypted = rsa_decrypt(encrypted, d, n);
    printf("Decrypted message: %d\n", decrypted);

    return 0;
}
